torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -122,6 +123,17 @@ TensorView::TensorView(const TensorView* src, IrCloner* ir_cloner)
   }
 }
 
+// TODO: Remove, only used for lowering
+TensorView::TensorView(
+    IrBuilderPasskey passkey,
+    const fuser::cuda::TensorView* tv)
+    : Val(passkey, ValType::TensorView, tv->getDataType().value()),
+      fuser_tv_(tv) {
+  setName(passkey, tv->name());
+  domain_ = GpuLower::current()->lowerValue(tv->domain())->as<TensorDomain>();
+  memory_type_ = tv->getMemoryType();
+}
```

Comment:
I'm assuming this is a temporary function, but I'm a little concerned about mixing the FIR and KIR TensorViews since some of the data fields of the `TensorView` class are not valid when it's KIR. For example, `TensorView::getComputeAtPosition()` is invalid when the TensorView object is KIR. It should never be called with KIR TensorViews, so having runtime assertions would be safer just to avoid mistakes.

This actually doesn't seem to be specific to TensorView, but now that more classes unify FIR and KIR, it seems it should be considered a best practice for each of IR node functions to have an assertion using `isKirStmt` whenever it makes sense.

Reply:
I'll try to take a pass at getting more consistent assertions is. It is all temporary and we want all information during lowering present in fusion